mcos/error_estimator.py

```diff
@@ -22,11 +22,28 @@ class ExpectedOutcomeErrorEstimator(AbstractErrorEstimator):
     """Error Estimator that calculates the mean difference in expected outcomes"""
 
     def estimate(self, mu: np.array, cov: np.array, allocation: np.array, optimal_allocation: np.array) -> float:
-        return np.dot(optimal_allocation - allocation, mu)
+        return _mean_difference_expected_outcome(optimal_allocation, allocation, mu)
 
 
 class VarianceErrorEstimator(AbstractErrorEstimator):
     """Error Estimator that calculates the mean difference in variance"""
 
     def estimate(self, mu: np.array, cov: np.array, allocation: np.array, optimal_allocation: np.array) -> float:
-        return np.dot(optimal_allocation - allocation, np.dot(cov, optimal_allocation - allocation))
+        return _mean_difference_variance(cov, allocation, optimal_allocation)
+
+
+class SharpeRatioErrorEstimator(AbstractErrorEstimator):
+    """Error estimator that calculates the mean difference in Sharpe ratio"""
+
+    def estimate(self, mu: np.array, cov: np.array, allocation: np.array, optimal_allocation: np.array) -> float:
+        mean_difference_expected_outcome = _mean_difference_expected_outcome(optimal_allocation, allocation, mu)
+        mean_difference_variance = _mean_difference_variance(cov, allocation, optimal_allocation)
+        return mean_difference_expected_outcome / mean_difference_variance ** .5
```

Comment:
I'm not sure the private functions are necessary. We could just re-use the error estimators. i.e.
```python
        mean_difference_expected_outcome = ExpectedOutcomeErrorEstimator().estimate(optimal_allocation, allocation, mu)
        mean_difference_variance = VarianceErrorEstimator().estimate(optimal_allocation, allocation, mu)
        return mean_difference_expected_outcome / mean_difference_variance ** .5
```

Reply:
I think it makes more sense to have the private functions since these are formulas, initializing the 2 classes and calling `.estimate()` makes it less clear that it's just one equation divided by the square root of another. I can change it if you'd like though.

Reply:
Makes sense! I don't have a super strong preference

Reply:
Could you change the ` ** .5` to `np.sqrt` instead? There's a speed gain with the latter (though it's trivial), and it makes the order of operations more explicit.